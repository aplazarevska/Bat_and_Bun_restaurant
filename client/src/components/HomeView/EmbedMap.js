import BingMapsReact from "bingmaps-react";
import { useMediaQuery } from '@mui/material';

const EmbedMap = () => {
  const matches = useMediaQuery('(min-width:900px)');

  return (
    <section id="embed-map">
      <BingMapsReact
        bingMapsKey={process.env.REACT_APP_BINGMAP_KEY}
        height={matches ? "500px" : "300px"}
        mapOptions={{
          navigationBarMode: "square",
        }}
        width="100%"
        viewOptions={{
          center: { latitude: -33.883320, longitude: 151.218384 },
          zoom: 17
        }}
        pushPins={[{ center: { latitude: -33.883320, longitude: 151.218384 }, options: { title: "Bat + Bun", color: '#3162ae' } }]}
      />
    </section>
  )
}

export default EmbedMap;
