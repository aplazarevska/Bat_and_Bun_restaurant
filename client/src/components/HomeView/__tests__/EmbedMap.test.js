import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import EmbedMap from '../EmbedMap';
import { afterEach, expect } from '@jest/globals';

const imgUrl = "./img/bat-bun-map.jpg";

afterEach(() => {
  cleanup();
});

test("Should render embed map component that has a background image", () => {
  render(<EmbedMap imgUrl={imgUrl} />);
  const embedMapElement = screen.getByTestId('embed-map');
  expect(embedMapElement).toBeInTheDocument();
  expect(embedMapElement).toHaveStyle(`background-image: url(${imgUrl})`);
});

test("Matches snapshot", () => {
  const tree = renderer.create(<EmbedMap imgUrl={imgUrl} />).toJSON();
  expect(tree).toMatchSnapshot();
});
