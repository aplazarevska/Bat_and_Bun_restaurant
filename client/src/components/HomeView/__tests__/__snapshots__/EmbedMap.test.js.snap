// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  data-testid="embed-map"
  id="embed-map"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(./img/bat-bun-map.jpg)",
    }
  }
/>
`;
